core/controllers/collection_editor_test.py

```diff
@@ -99,6 +109,85 @@ def test_access_collection_editor_page(self):
         self.assert_can_edit(response.body)
         self.logout()
 
+    def test_editable_collection_handler_get(self):
+        whitelisted_usernames = [self.EDITOR_USERNAME]
+        self.set_config_property(
+            config_domain.WHITELISTED_COLLECTION_EDITOR_USERNAMES,
+            whitelisted_usernames)
+
+        # Check that non-editors cannot access the editor data handler.
+        # This is due to them not being whitelisted.
+        response = self.testapp.get(
+            '%s/%s' % (feconf.EDITABLE_COLLECTION_DATA_URL_PREFIX,
+                       self.COLLECTION_ID))
+        self.assertEqual(response.status_int, 302)
+
+        # Check that whitelisted users can access the data
+        # from the editable_collection_data_handler
+        self.login(self.EDITOR_EMAIL)
+
+        json_response = self.get_json(
+            '%s/%s' % (feconf.EDITABLE_COLLECTION_DATA_URL_PREFIX,
+                       self.COLLECTION_ID))
+        self.assertEqual(self.COLLECTION_ID, json_response['collection']['id'])
+        self.assertEqual(
+            'Introduction to Collections in Oppia',
+            json_response['collection']['title'])
+        self.logout()
+
+    def test_editable_collection_handler_put_cannot_access(self):
+        """Check that non-editors cannot access editable put handler"""
+        whitelisted_usernames = [self.EDITOR_USERNAME]
+        self.set_config_property(
+            config_domain.WHITELISTED_COLLECTION_EDITOR_USERNAMES,
+            whitelisted_usernames)
+
+        # Check that Non_whitelisted users cannot access the put method
+        # from the editable_collection_data_handler.
+        self.login(self.OWNER_EMAIL)
```

Comment:
I believe I have added the tests you were thinking of, ensuring editors have access and viewers (Non-editors) do not have access. I ended up modifying the can_access test as well. Let me know if this is more along the lines of what you were thinking.